### PR TITLE
feat/group-filter Add PHPUnit Group Filtering & Run Support

### PR DESCRIPTION
**Pull request by Jhuntenburg**
### Summary

This PR adds support for **PHPUnit test groups**, enabling users to **filter and run tests by group** directly from the VS Code Test Explorer.

It supports both traditional `@group` annotations and PHP 8+ `#[Group]` attributes, mapping them into Test Explorer tags and providing a new command to execute grouped tests in a single PHPUnit process.

> ⚠️ **Note:** TypeScript is not my primary language. While this feature is working in practice (see demo video below), I’d really appreciate a careful review of the TS structure, patterns, and edge cases.

---

#### Group Parsing
- Parses `@group` annotations and `#[Group('name')]` attributes
- Works at both class and method level

#### Test Explorer Integration
- Groups appear as filterable tags in the VS Code Test Explorer
- Filter syntax:
@phpUnitTestController:group:

#### Run Tests by Group
- New command: **PHPUnit: Run tests by group**
- Displays a picker UI to select a group and run all matching tests

#### Tag Inheritance
- Test methods inherit group tags from their parent class
- Filtering by a class-level group automatically includes all its test methods

#### Single Process Execution
- Uses PHPUnit’s native `--group` flag
- Runs all matching tests in a single PHPUnit process
- Avoids database and container conflicts caused by parallel execution

---

### Demo

📽️ **Video walkthrough included**, demonstrating:
- Group tags appearing in the Test Explorer
- Filtering tests by group
- Running a group via the command palette

https://youtu.be/WJf1DIRIcNI

---

### Files Changed

- `src/PHPUnit/TestParser/AnnotationParser.ts`  
→ Added group parsing support

- `src/PHPUnit/types.ts`  
→ Added `group?: string[]` to `Annotations`

- `src/TestCollection/TestHierarchyBuilder.ts`  
→ Added `GroupRegistry` singleton  
→ Assigned group tags with parent → child inheritance

- `src/CommandHandler.ts`  
→ New `runByGroup()` command and picker UI

- `src/Handler.ts`  
→ New `startGroupTestRun()` using PHPUnit `--group`

- `src/extension.ts`  
→ Command registration

- `package.json`  
→ Command contribution

- `src/PHPUnit/TestParser/PHPUnitParser.test.ts`  
→ Unit tests for group parsing

### CODE REVIEW

File: src/extension.test.ts [hunk 1]
```diff
@@ -11,7 +11,8 @@ import { Configuration } from './Configuration';
 import { activate } from './extension';
 import { getPhpUnitVersion, getPhpVersion, normalPath, pestProject, phpUnitProject } from './PHPUnit/__tests__/utils';
 
-jest.mock('child_process');
+//updated to match spawn for tests
+jest.mock('node:child_process');
```
Comment (Jhuntenburg) [inline src/extension.test.ts]: This change was because i was getting test failures from run npm jest, this change fixed on local, not sure if you were haveing the same issue.

**Comment by recca0120** [comment]
cool, merge later
